sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -140,6 +140,18 @@ class FileStreamSourceSuite extends FileStreamSourceTest with SharedSQLContext {
 
   import testImplicits._
 
+  private def withSchemaInference(f: => Unit): Unit = {
+    withSQLConf("spark.sql.streaming.schemaInference" -> "true")(f)
```

Comment:
Can we use the key here so if we change it there will be a compile error?  I'd also consider just inlining these to make the test cases more self contained.